crates/iota-genesis-builder/src/stardust/process_outputs.rs

```diff
@@ -67,9 +81,253 @@ pub fn is_participation_output(output: &Output) -> bool {
+                                    );
+                                    continue;
+                                }
+                            } else {
+                                // if it is just a basic output, try to perform the SwapSplit
+                                // operation for several destinations once all tokens targets are
+                                // meet.
+                                let (original_output_opt, split_outputs) = swap_split_operation(
+                                    destinations.iter_by_tokens_target_mut_filtered(),
+                                    basic_output,
+                                );
+                                // if some SwapSplit were performed, their result are basic inputs
+                                // stored in split_outputs; so, we save them in
+                                // split_basic_outputs to return them later
+                                if !split_outputs.is_empty() {
+                                    self.num_swapped_basic += 1;
+                                }
+                                self.split_basic_outputs.extend(
+                                    split_outputs
+                                        .into_iter()
+                                        .map(|output| (header.clone(), output)),
+                                );
+                                // if there was a remainder, the original output is returned for the
+                                // iterator, possibly with a modified amount; else, continue the
+                                // loop
+                                if let Some(original_output) = original_output_opt {
+                                    *inner = original_output;
+                                } else {
+                                    continue;
+                                }
+                            };
+                        }
+                    }
+                }
+            }
+            return Some(output);
+        }
+        // Now that we are out of the loop we collect the processed outputs from the
+        // timelock filter and try to fulfill the target.
+        // First, resolve timelocks SwapSplit operations, taking those out from
+        // timelocks; the ordered_timelock_candidates is ordered by timestamp
+        // and we want to take the latest ones first.
+        while let Some((header, mut inner)) = self.ordered_timelock_candidates.pop() {
+            // We know that all of them are timelocked basic outputs
+            let timelocked_basic_output = inner.as_basic();
+            let uc = timelocked_basic_output.unlock_conditions();
+            // Get destination address and mutable timelocked tokens target
+            if let Some(destinations) = self
+                .swap_split_map
+                .get_destination_maybe_mut(uc.address().unwrap().address())
```

Comment:
Nit: We should expect this to be `Some`, no?

```suggestion
            let destinations = self
                .swap_split_map
                .get_destination_maybe_mut(uc.address().unwrap().address())
                .expect("ordered timelock candidates should be part of the swap map");
```

Reply:
Nope, for the majority of the outputs that `get_destination_maybe_mut ` will return `None`, because we are looking for all and only outputs with address UC being the one indicated in the address swap map CSV.

Reply:
But isn't it necessary for an output to be part of `self.timelock_candidates` to have passed the condition in line 203? Probably I miss something.

Reply:
My bad, I thought the comment was related to the `get_destination_maybe_mut` call above in line 203, you're right!